DopplerCurrencyJob/DopplerCurrencyJob.cs

```diff
@@ -25,27 +29,98 @@ public class DopplerCurrencyJob
             _dopplerSapService = dopplerSapService;
         }
 
+
         [AutomaticRetry(OnAttemptsExceeded = AttemptsExceededAction.Delete, Attempts = 0)]
-        public IList<CurrencyResponse> Run() => RunAsync().GetAwaiter().GetResult();
+        public DopplerCurrencyJobResponse Run() => RunAsync().GetAwaiter().GetResult();
 
-        private async Task<IList<CurrencyResponse>> RunAsync()
+        private async Task<DopplerCurrencyJobResponse> RunAsync()
         {
-            _logger.LogInformation("Getting currency per each code enabled.");
-            var currencyDto = await _dopplerCurrencyService.GetCurrencyByCode();
+            var taskFailed = false;
+
+            var resultBuffer = new StringBuilder();
+            var message = "Getting currency per each code enabled.";
+            _logger.LogInformation(message);
+            resultBuffer.Append(message).AppendLine();
 
+            var currencyDto = await _dopplerCurrencyService.GetCurrencyByCode();
             if (!currencyDto.Any())
             {
-                _logger.LogInformation("Non-existent currencies for this date, please check for errors.");
-                return new List<CurrencyResponse>();
+                message = "Non-existent currencies for this date, please check for errors.";
+                _logger.LogInformation(message);
+                resultBuffer.Append(message).AppendLine();
+                return new DopplerCurrencyJobResponse()
+                {
+                    CurrencyList = new List<CurrencyResponse>(),
+                    Message = resultBuffer.ToString()
+                };
             }
 
-            _logger.LogInformation("Sending currency data to Doppler SAP system.");
-            await _dopplerSapService.SendCurrency(currencyDto);
+            message = "Currency retrieved by code:";
+            _logger.LogInformation(message);
+            resultBuffer.Append(message).AppendLine();
+            resultBuffer.Append(JsonSerializer.Serialize(currencyDto)).AppendLine();
```

Comment:
This could be also simplified in the same line:

```suggestion
            AppendSerializedToStringBuilder(currencyDto);
```

Reply:
Hecho, marco como resuelto.